testing/mocks/forest.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Optakt Labs OÜ
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+package mocks
+
+import (
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type Forest struct {
+	SaveFunc   func(tree *trie.MTrie, paths []ledger.Path, parent flow.StateCommitment)
+	HasFunc    func(commit flow.StateCommitment) bool
+	TreeFunc   func(commit flow.StateCommitment) (*trie.MTrie, bool)
+	PathsFunc  func(commit flow.StateCommitment) ([]ledger.Path, bool)
+	ParentFunc func(commit flow.StateCommitment) (flow.StateCommitment, bool)
+	ResetFunc  func(finalized flow.StateCommitment)
+	SizeFunc   func() uint
```

Comment:
This makes me think I should remove the size function. It's not relevant for functionality.